flight/Modules/VtolPathFollower/vtol_follower_fsm.c

```diff
@@ -80,7 +80,7 @@ enum vtol_fsm_event {
  * sets the nav mode and a default methods will farm it out to the appropriate
  * algorithm.
  */
-enum vtol_fsm_state {
+enum vtol_fsm_state_num {
```

Comment:
The old names were consistent with other state machines in the code:
https://github.com/TauLabs/TauLabs/blob/next/flight/PiOS/STM32F4xx/inc/pios_i2c_priv.h#L53
https://github.com/TauLabs/TauLabs/blob/next/flight/PiOS/STM32F4xx/pios_i2c.c#L80

I added the static_fn to the current state structure (hence making the name _transition) slightly less accurate. My preference would be to keep these names the same and consistent with other FSM's in our code (I think @solidgoldbomb took this convention/design pattern from the linux kernel).

Not a blocker though if you really want to change it :)